pytaskflow/server/worker.py

```
@@ -129,14 +129,20 @@ def run(self):
             try:
                 self._run_schedulers()
                 # 1. Dequeue a job
-                job = self.storage.dequeue(self.queues, timeout_seconds=1)
-                
-                if job:
-                    logger.info(f"[{self.worker_id}] Picked up job {job.id}")
+                dequeued_job = self.storage.dequeue(self.queues, timeout_seconds=0.1)
+                if dequeued_job:
+                    # Fetch the latest job data from storage to ensure retry_count is up-to-date
+                    job = self.storage.get_job_data(dequeued_job.id)
+                    if not job: # Job might have been deleted or moved by another process
```

Flake8 (7.2.0)

[error] at least two spaces before inline comment

(E261)